### PR TITLE
131 allow pasting lists of email addresses into recipient fields

### LINKED ISSUE #131: Allow pasting multiple addresses in recipient fields
When pasting several email addresses in a recipient field (delimited by a space, a comma or a semicolon) these should be accepted as separate addresses instead of prompting the error “Please enter a valid email address”.

Ref: https://community.runbox.com/t/pasting-several-email-addresses-in-the-to-field/816

### PR DESCRIPTION
**Pull request by castaway**
Fixes #131 

Also refactors:
* MailAddressInfo out of messageinfo
* addRecipient* shared code into one method

### CODE REVIEW

File: src/app/common/mailaddressinfo.spec.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+import { MailAddressInfo } from '../common/mailaddressinfo';
+
+describe('MailAddressInfo', () => {
+    it('Construct with name and email', () => {
+        const ma = new MailAddressInfo('Test', 'test1@runbox.com');
+        // expect(ma.name).toBe('Test');
```
Comment (tadzik) [inline src/app/common/mailaddressinfo.spec.ts:25]: Hm, shouldn't these be either uncommented or removed?

File: src/app/common/mailaddressinfo.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+export class MailAddressInfo {
+  nameAndAddress: string;
```
Comment (tadzik) [inline src/app/common/mailaddressinfo.ts:21]: Minor: indentation looks off :)

File: src/app/compose/mailrecipientinput.component.html [hunk 3]
```diff
@@ -11,12 +11,11 @@
                 [matChipInputSeparatorKeyCodes]="separatorKeysCodes"                
                 [matChipInputAddOnBlur]="false"
                 (blur)="addRecipientFromBlur()"
-                (matChipInputTokenEnd)="addRecipient($event)" 
+                (matChipInputTokenEnd)="addRecipientFromEnter($event)" 
                 [matAutocomplete]="auto" style="flex-grow: 1" 
-                type="email"
-                email
+               type="email"
```
Comment (tadzik) [inline src/app/compose/mailrecipientinput.component.html:16]: Minor: Looks like a small indent fail (at least here on github).

File: src/app/common/mailaddressinfo.spec.ts (hunk 1 above)
Comment (castaway) [reply src/app/common/mailaddressinfo.spec.ts:25]: Not sure what you mean, these are new tests..

Comment (tadzik) [reply src/app/common/mailaddressinfo.spec.ts:25]: Right – shouldn't they be ran then? ;)

Comment (castaway) [reply src/app/common/mailaddressinfo.spec.ts:25]: Oh! those lines, yes.. true..

**Pull request merged**